src/plot_api/helpers.js

```diff
@@ -102,13 +102,12 @@ exports.cleanLayout = function(layout) {
         }
     }
 
-    if(layout.annotations !== undefined && !Array.isArray(layout.annotations)) {
```

Comment:
this commit was enough to fix bug discovered in http://codepen.io/etpinard/pen/xRGgwW